Program.cs

```diff
@@ -1,12 +1,261 @@
+        private string _personSurname;
+        private string _personName;
+        private int _personAge;
+
+        public string PersonSurname
+        {
+            set { SetSurname(_personSurname = value); }
+            get { return GetSurname(); }
+        }
+
+        public string PersonName
+        {
+            set { SetName(_personName = value); }
+            get { return GetName(); }
+        }
+
+        public int PersonAge
+        {
+            set
+            {
+                SetAge(_personAge = value);
+            }
+            get
+            {
+                return GetAge();
+            }
+        }
+
+        public void SetSurname(string personSurname)
+        {
+            _personSurname = personSurname;
+        }
+
+        public void SetName(string personName)
+        {
+            _personName = personName;
+        }
+
+        public void SetAge(int personAge)
+        {
+            _personAge = personAge;
+        }
+
+        public void WritePersonData()
+        {
+            Console.WriteLine($"{_personSurname} {_personName}, {_personAge}");
+        }
+
+        public void Print()
+        {
+            Console.WriteLine($"{_personSurname} {_personName}, {_personAge} y.o.");
+        }
+
+        public string GetSurname()
+        {
+            return _personSurname;
+        }
+
+        public string GetName()
+        {
+            return _personName;
+        }
+
+        public int GetAge()
+        {
+            return _personAge;
+        }
+    }
+}
+
+namespace Exercises
+{
+    public class Student : Person, IPrintable
+    {
+        private string _studentAbility;
+        private int _studentAssesment;
+
+        public string StudentAbility => _studentAbility;
+
+        public int StudentAssesment => _studentAssesment;
+
+        public void SetStudentAbility(string studentAbility)
+        {
+            _studentAbility = studentAbility;
+        }
+
+        public void SetStudentAssesment(int assesment)
+        {
+            _studentAssesment = assesment;
+        }
+    }
+}
+
+namespace Exercises
+{
+    public class Teacher : Person, IPrintable
+    {
+        private string _teacherTemperament;
+        private string _teacherSubject;
+
+        public string TeacherTemperament => _teacherTemperament;
+
+        public string TeacherSubject => _teacherSubject;
+
+        public void SetTeacherTemperament(string teacherTemperament)
+        {
+            _teacherTemperament = teacherTemperament;
+        }
+
+        public void SetTeacherSubject(string subject)
+        {
+            _teacherSubject = subject;
+        }
+    }
+}
+
+namespace Exercises
+{
+    public class Classroom : IPrintable
+    {
+        private List<Person> _persons = new List<Person>();
+        private Student _student;
+        private Teacher _teacher;
+
+        public void AddPersons()
+        {
+            _persons.Add(_teacher);
+            _persons.Add(_student);
```

Comment:
All links will be added here, because... there is nowhere for people to come from